app/src/ui/lib/update-store.ts

```diff
@@ -160,7 +160,7 @@ class UpdateStore {
     // the arm64 binary.
     if (
       enableUpdateFromEmulatedX64ToARM64() &&
-      (remote.app.runningUnderRosettaTranslation === true ||
+      ((await isRunningUnderRosettaTranslation()) === true ||
         isRunningUnderARM64Translation() === true)
```

Comment:
FYI Electron 15 adds my contribution of the [`runningUnderARM64Translation`](https://www.electronjs.org/docs/latest/api/app#apprunningunderarm64translation-readonly-macos-windows) property which unifies these two checks.